api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -124,6 +128,18 @@ protected static boolean validateField10Digits(StringBuilder errors, String fiel
         return false;
     }
 
+    protected static boolean validatetypeASHA(StringBuilder errors, String fieldName, String mctsFlwId, Long contactNumber, String type) {
```

Comment:
I think I understand where the confusion is now. The bug primarily only talks about filtering asha users for import. And this validation exists in the controller, which is used only for incoming/outgoing calls to IMI.

If we are only dealing with imports, this is not the right place for the validation.
If we are only dealing with live call filtering, then this is ok and you would add these associated check to the UserController